embassy-stm32/src/rng.rs

```diff
@@ -19,13 +19,75 @@ impl<T: Instance> Random<T> {
 use embassy::traits::rng::Rng as RngTrait;
 use core::future::Future;
 use core::marker::PhantomData;
+use core::task::{Context, Poll};
+use core::pin::Pin;
+
+static RNG_WAKER: AtomicWaker = AtomicWaker::new();
+const RNG: pac::rng_v1::Rng = pac::rng_v1::Rng( <peripherals::RNG as sealed::Instance>::ADDR as _);
+
+pub unsafe fn on_irq() {
+    if is_ready() || is_seed_error() || is_clock_error() {
```

Comment:
This'll read `sr` 3 times (compiler won't coalesce reads because they're volatile). It's nicer to do just 1 read then check the individual fields